src/redux/books/booksSlice.js

```diff
@@ -1,9 +1,29 @@
 /* eslint-disable no-param-reassign */
```

Comment:
Well done overall 💯 I read your comment on the previous review and this one, but while doing some research, I found a workaround allowing you to do it with all `eslint` rules enabled, the code is still working 🔍 . First, you have to understand where the error is coming from. This ESLint error is caused by the fact that you are trying to modify the state object directly inside the reducer functions 👀 , which is generally not recommended. In Redux, the state is considered immutable and any updates to it should be done in an immutable way.

To fix this error, you can create a new state object based on the previous state and the changes you want to make. In your `addBook` reducer, you can create a new array that includes the previous books and the new book you want to add, like this:
```
addBook: (state, action) => {
  const newBook = {
    item_id: uuidv4(),
    title: action.payload.title,
    author: action.payload.author,
    category: action.payload.category,
  };
  return { ...state, books: [...state.books, newBook] };
},
```
Similarly, in your `removeBook` reducer, you can create a new array that excludes the book you want to remove, like this:
```
removeBook: (state, action) => {
  const bookId = action.payload;
  return { ...state, books: state.books.filter((book) => book.item_id !== bookId) };
},
```
By creating a new state object this way, you are ensuring that the original state object remains unchanged, and the changes you make are done in an immutable way, which is recommended in Redux. 🤓

Reply:
Wow, thank you, @didierganthier I can't believe it was that simple, I truly appreciate the commitment you've put into finding a working solution, I owe you one. 💪

Reply:
Anytime man 😉 🚀